src/main/java/mServer/crawler/sender/br/Resolution.java

```diff
@@ -0,0 +1,153 @@
+
+  private final String description;
+
+  private Resolution(final int resolutionSize, final String description) {
+    this.resolutionSize = resolutionSize;
+    this.description = description;
+  }
+
+  public static List<Resolution> getFromBestToLowest() {
+    return Arrays.asList(Resolution.values()).stream()
+        .sorted(Comparator.comparing(Resolution::getResolutionSize).reversed())
+        .collect(Collectors.toList());
+  }
+
+  public static Resolution getHighestResolution() {
+    return Resolution.HD;
+  }
+
+  public static Resolution getLowestResolution() {
+    return Resolution.VERY_SMALL;
+  }
+
+  public static Resolution getNextHigherResolution(final Resolution startingResolution) {
+    return getNextResolutionByDirection(startingResolution, CountingDirection.HIGHER);
+  }
+
+  public static Resolution getNextLowerResolution(final Resolution startingResolution) {
+    return getNextResolutionByDirection(startingResolution, CountingDirection.LOWER);
+  }
+
+  /**
+   * Derzeit sind folgende ARD AudioVideo Ordinals bekannt:<br>
+   * <ul>
+   * <li>HD = 1280 width</li>
+   * <li>Premium = 969 width</li>
+   * <li>Large = 640 width</li>
+   * <li>Standard = 512 width</li>
+   * <li>Mobile = 480 width</li>
+   * <li>Mobile_S = 320 width</li>
+   * </ul>
+   *
+   * @param profileName
+   * @return
+   */
+  public static Resolution getResolutionFromArdAudioVideoOrdinalsByProfileName(
+      final String profileName) {
+    if (profileName.endsWith("HD")) {
+      return Resolution.HD;
+    }
+    if (profileName.endsWith("Premium")) {
+      return Resolution.HD;
+    }
+    if (profileName.endsWith("Large")) {
+      return Resolution.NORMAL;
+    }
+    if (profileName.endsWith("Standard")) {
+      return Resolution.SMALL;
+    }
+    if (profileName.endsWith("Mobile")) {
+      return Resolution.SMALL;
+    }
+
+    return Resolution.VERY_SMALL;
+
+  }
+
+  /**
+   * The following width size limits are relevant:<br>
+   * <ul>
+   * <li>HD = >= 969 width</li>
+   * <li>Normal = >= 640 width</li>
+   * <li>Small = >= 512 width</li>
+   * <li>Very Small = < 512 width</li>
+   * </ul>
+   *
+   * @param width
+   * @return
+   */
+  public static Resolution getResolutionFromWidth(final int width) {
+    if (width >= 969) {
+      return Resolution.HD;
+    }
+    if (width >= 640) {
+      return Resolution.NORMAL;
+    }
+    if (width >= 512) {
+      return Resolution.SMALL;
+    }
+    return Resolution.VERY_SMALL;
+  }
+
+  static Resolution getNextResolutionByDirection(final Resolution startingResolution,
+      final CountingDirection direction) {
+    try {
+      return getResoultionByResolutionSize(
+          startingResolution.getResolutionSize() + direction.getDirectionValue());
+    } catch (final NoSuchElementException nsee) {
+      return startingResolution;
+    }
+  }
+
+
+  static Resolution getResoultionByResolutionSize(final int searchedResolutionSize)
+      throws NoSuchElementException {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'java.util.NoSuchElementException' which is a runtime exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)